nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -242,11 +242,14 @@ let
           deviceDependency = if (config.boot.isContainer || i.name == "lo")
             then []
             else [ (subsystemDevice i.name) ];
+          dhcpDependency = if i.macAddress == null
+            then []
+            else [ "dhcpcd.service" ];
         in
         nameValuePair "network-link-${i.name}"
         { description = "Link configuration of ${i.name}";
           wantedBy = [ "network-interfaces.target" ];
-          before = [ "network-interfaces.target" ];
+          before = [ "network-interfaces.target" ] ++ dhcpDependency;
```

Comment:
eeeh… This is trying to mask a problem which we wouldn't have if we'd apply interface parameters as soon as the interfaces appear, via udev. Even if we'd merge this PR as-is and add this dependency, things could still fail during reconfiguration.

I made the scripts below scripted-networking only in https://github.com/NixOS/nixpkgs/pull/85170, as running these scripts for networkd definitely was wrong, and I was too lazy to think about how to make scripted networking better, as I mostly use the networkd codepaths, 

However, mac address and MTU should be perfectly fine configurable via a `systemd.network.links` directive even in the non-networkd case (as it's handled by udev, not networkd) since https://github.com/NixOS/nixpkgs/pull/82941.

The only line in the script that would need to be left would be the line bringing up the interface.